WelsonJS.Toolkit/WelsonJS.Launcher/ResourceTools/ImageColorPicker.cs

```diff
@@ -0,0 +1,343 @@
+                    return;
+                }
+
+                if (!TryParseJsonInt(xJson, out var x) || !TryParseJsonInt(yJson, out var y))
+                {
+                    context.Response.StatusCode = (int)HttpStatusCode.BadRequest;
+                    await WriteJsonAsync(context, new Dictionary<string, object>
+                    {
+                        ["ok"] = false,
+                        ["error"] = "'point.x' and 'point.y' must be JSON numbers."
+                    });
+                    return;
+                }
+
+                // Allow data URL prefix (data:*;base64,....)
+                imageB64 = StripDataUrlPrefix(imageB64);
+
+                byte[] bytes;
+                try { bytes = Convert.FromBase64String(imageB64); }
+                catch
+                {
+                    context.Response.StatusCode = (int)HttpStatusCode.BadRequest;
+                    await WriteJsonAsync(context, new Dictionary<string, object>
+                    {
+                        ["ok"] = false,
+                        ["error"] = "Invalid base64 image data."
+                    });
+                    return;
+                }
+
+                // ---- Decode image and sample pixel --------------------------------------------
+                using (var ms = new MemoryStream(bytes, writable: false))
+                using (var bmp = new Bitmap(ms))
+                {
+                    if (x < 0 || y < 0 || x >= bmp.Width || y >= bmp.Height)
+                    {
+                        context.Response.StatusCode = (int)HttpStatusCode.BadRequest;
+                        await WriteJsonAsync(context, new Dictionary<string, object>
+                        {
+                            ["ok"] = false,
+                            ["error"] = $"Point out of bounds. Image size: {bmp.Width}x{bmp.Height}, requested: ({x},{y})."
+                        });
+                        return;
+                    }
+
+                    var color = bmp.GetPixel(x, y);
+                    var hex = $"#{color.R:X2}{color.G:X2}{color.B:X2}";
+                    var name = ToCommonColorName(color); // null if not close to a known color
+
+                    context.Response.StatusCode = (int)HttpStatusCode.OK;
+                    await WriteJsonAsync(context, new Dictionary<string, object>
+                    {
+                        ["ok"] = true,
+                        ["hex"] = hex,
+                        ["name"] = name, // may be null → serializer will emit "name": null
+                        ["rgb"] = new Dictionary<string, object>
+                        {
+                            ["r"] = color.R,
+                            ["g"] = color.G,
+                            ["b"] = color.B
+                        }
+                    });
+                }
+            }
+            catch (Exception ex)
+            {
+                _logger?.Error($"ImageColorPicker error: {ex}");
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Avoid string interpolation within log messages, as it can blur the lines between variables and the log content. Instead, opt for structured logging, where variables are passed as separate arguments. This improves the clarity of logs and facilitates efficient log analysis. For more information, refer to https://github.com/NLog/NLog/wiki/How-to-use-structured-logging.